packages/web3-eth-contract/test/integration/contract_deploy.test.ts

```diff
@@ -41,6 +51,38 @@ describe('contract', () => {
 
 			sendOptions = { from: acc.address, gas: '1000000' };
 		});
+		describe('local account', () => {
+			it.each([signTxAndSendEIP1559, signTxAndSendEIP2930])(
+				'should deploy the contract %p',
+				async signTxAndSend => {
+					const deployData = contract.deploy(deployOptions);
+
+					const res = await signTxAndSend(
+						{
+							data: deployData.encodeABI(),
+						},
+						localAccount.privateKey,
+					);
+					expect(Number(res.status)).toBe(1);
```

Comment:
after deployment also check that new contract instance address is set
`contractInstance.options.address`

Reply:
done